### PR TITLE
gum: init at 0.1.0

### PR DESCRIPTION
**Pull request by maaslalani**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by pbsds — APPROVED** [review]

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 183396` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gum</li>
  </ul>
</details>

**Comment by maaslalani** [comment]
> I was in the process of packaging it myself when i found this PR.

Thanks so much @pbsds, really appreciate it ❤️

**Comment by shadowrylander** [comment]
Hello!

Sorry, I got a little impatient! 😅

When I use your derivation, I'm getting this error:
```
go test proxy running at GOPROXY=http://127.0.0.1:44259/mod
--- FAIL: TestScript (0.11s)
    --- FAIL: TestScript/mod_find (0.32s)
        script_test.go:253: 
            # Derive module path from import comment. (0.026s)
            # Import comment works even with CRLF line endings. (0.058s)
            # Derive module path from location inside GOPATH.
            # 'go mod init' should succeed if modules are not explicitly disabled. (0.028s)
            # go mod init rejects a zero-length go.mod file (0.086s)
            # Module path from Godeps/Godeps.json overrides GOPATH. (0.064s)
            # Empty directory outside GOPATH fails. (0.024s)
            # Empty directory inside GOPATH/src uses location inside GOPATH. (0.008s)
            # In Plan 9, directories are automatically created in /n.
            # For example, /n/go.mod always exist, but it's a directory.
            # Test that we ignore directories when trying to find go.mod. (0.018s)
            > cd $WORK/gomoddir
            $WORK/gomoddir
            > ! go list .
            [stderr]
            go: cannot find main module, but found .git/config in /
                to create a module there, run:
                cd ../../../../../.. && go mod init
            [exit status 1]
            > stderr '^go: go.mod file not found in current directory or any parent directory; see ''go help modules''$'
            FAIL: testdata/script/mod_find.txt:52: no match for `(?m)^go: go.mod file not found in current directory or any parent directory; see 'go help modules'$` found in stderr
            
    --- FAIL: TestScript/mod_convert_dep (0.04s)
        script_test.go:253: 
            # We should not create a go.mod file unless the user ran 'go mod init' explicitly.
            # However, we should suggest 'go mod init' if we can find an alternate config file. (0.011s)
            # The command we suggested should succeed. (0.020s)
            # In Plan 9, directories are automatically created in /n.
            # For example, /n/Gopkg.lock always exists, but it's a directory.
            # Test that we ignore directories when trying to find alternate config files. (0.010s)
            > cd $WORK/gopkgdir/x
            $WORK/gopkgdir/x
            > ! go list .
            [stderr]
            go: cannot find main module, but found .git/config in /
                to create a module there, run:
                cd ../../../../../../.. && go mod init
            [exit status 1]
            > stderr '^go: go.mod file not found in current directory or any parent directory; see ''go help modules''$'
            FAIL: testdata/script/mod_convert_dep.txt:21: no match for `(?m)^go: go.mod file not found in current directory or any parent directory; see 'go help modules'$` found in stderr
            
FAIL
FAIL    cmd/go  190.854s
```

**Comment by dit7ya** [comment]
Result of `nixpkgs-review pr 183396` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gum</li>
  </ul>
</details>

**Review by dit7ya — APPROVED** [review]

**Comment by winterqt** [comment]
> When I use your derivation, I'm getting this error:

@shadowrylander Can you provide a full build log?

File: pkgs/applications/misc/gum/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    installManPage gum.1
+    installShellCompletion --cmd gum \
+      --bash <($out/bin/gum completion bash) \
+      --fish <($out/bin/gum completion fish) \
+      --zsh <($out/bin/gum completion zsh)
+  '';
+
+  meta = with lib; {
+    description = "Tasty Bubble Gum for your shell";
+    homepage = "https://github.com/charmbracelet/charm";
```
Comment (winterqt) [inline pkgs/applications/misc/gum/default.nix:33]:
```suggestion
    homepage = "https://github.com/charmbracelet/gum";
```

File: pkgs/applications/misc/gum/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{ lib, buildGoModule, installShellFiles, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "gum";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "charmbracelet";
+    repo = "gum";
```
Comment (winterqt) [inline pkgs/applications/misc/gum/default.nix:9]:
```suggestion
    repo = pname;
```

**Review by winterqt — CHANGES_REQUESTED** [review]

**Review by winterqt — APPROVED** [review]

**Comment by winterqt** [comment]
I'm going to merge this since the error @shadowrylander is running into looks to be unrelated to this package (in fact, it looks like it's related to the build of [Go itself](https://github.com/golang/go/blob/9a2001a8cc75b1ceda17da828edcee93c9d9f42c/src/cmd/go/testdata/script/mod_find.txt)). However, if you can still reproduce this error, please do provide a build log and reproduction steps.

**Comment by shadowrylander** [comment]
@winterqt Yep! Sorry for the delay; here you go!
```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/5vsadcpvc4a7s1hlf52b1iwk4snkaj05-go1.17.10.src.tar.gz
source root is go
setting SOURCE_DATE_EPOCH to timestamp 1652199429 of file go/test/zerodivide.go
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
patching script interpreter paths in ./
./misc/cgo/fortran/test.bash: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/iffl6dlplhv22i2xy7n1w51a5r631kmi-bash-5.1-p16/bin/bash"
./misc/wasm/go_js_wasm_exec: interpreter directive changed from "#!/bin/bash" to "/nix/store/iffl6dlplhv22i2xy7n1w51a5r631kmi-bash-5.1-p16/bin/bash"
./misc/arm/a: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/iffl6dlplhv22i2xy7n1w51a5r631kmi-bash-5.1-p16/bin/bash"
./misc/ios/clangwrap.sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/iffl6dlplhv22i2xy7n1w51a5r631kmi-bash-5.1-p16/bin/sh"
./src/cmd/compile/internal/ssa/gen/cover.bash: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/iffl6dlplhv22i2xy7n1w51a5r631kmi-bash-5.1-p16/bin/bash"
./src/cmd/go/mkalldocs.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/iffl6dlplhv22i2xy7n1w51a5r631kmi-bash-5.1-p16/bin/bash"
./src/cmd/go/testdata/failssh/ssh: interpreter directive changed from "#!/bin/sh" to "/nix/store/iffl6dlplhv22i2xy7n1w51a5r631kmi-bash-5.1-p16/bin/sh"
./src/all.bash: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/iffl6dlplhv22i2xy7n1w51a5r631kmi-bash-5.1-p16/bin/bash"
./src/internal/trace/mkcanned.bash: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/iffl6dlplhv22i2xy7n1w51a5r631kmi-bash-5.1-p16/bin/bash"
./src/race.bash: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/iffl6dlplhv22i2xy7n1w51a5r631kmi-bash-5.1-p16/bin/bash"
./src/make.bash: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/iffl6dlplhv22i2xy7n1w51a5r631kmi-bash-5.1-p16/bin/bash"
./src/run.bash: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/iffl6dlplhv22i2xy7n1w51a5r631kmi-bash-5.1-p16/bin/bash"
./src/buildall.bash: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/iffl6dlplhv22i2xy7n1w51a5r631kmi-bash-5.1-p16/bin/bash"
./src/syscall/mkall.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/iffl6dlplhv22i2xy7n1w51a5r631kmi-bash-5.1-p16/bin/bash"
./src/syscall/mksysctl_openbsd.pl: interpreter directive changed from "#!/usr/bin/env perl" to "/nix/store/dk5lnlbfz4vgyywifl4zaa988p9ixa2n-perl-5.34.1/bin/perl"
./src/syscall/mksysnum_plan9.sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/iffl6dlplhv22i2xy7n1w51a5r631kmi-bash-5.1-p16/bin/sh"
./src/syscall/mksysnum_freebsd.pl: interpreter directive changed from "#!/usr/bin/env perl" to "/nix/store/dk5lnlbfz4vgyywifl4zaa988p9ixa2n-perl-5.34.1/bin/perl"
./src/syscall/mksysnum_linux.pl: interpreter directive changed from "#!/usr/bin/env perl" to "/nix/store/dk5lnlbfz4vgyywifl4zaa988p9ixa2n-perl-5.34.1/bin/perl"
./src/syscall/mksysnum_dragonfly.pl: interpreter directive changed from "#!/usr/bin/env perl" to "/nix/store/dk5lnlbfz4vgyywifl4zaa988p9ixa2n-perl-5.34.1/bin/perl"
./src/syscall/mksyscall_libc.pl: interpreter directive changed from "#!/usr/bin/env perl" to "/nix/store/dk5lnlbfz4vgyywifl4zaa988p9ixa2n-perl-5.34.1/bin/perl"
./src/syscall/mksysnum_openbsd.pl: interpreter directive changed from "#!/usr/bin/env perl" to "/nix/store/dk5lnlbfz4vgyywifl4zaa988p9ixa2n-perl-5.34.1/bin/perl"
./src/syscall/mkerrors.sh: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/iffl6dlplhv22i2xy7n1w51a5r631kmi-bash-5.1-p16/bin/bash"
./src/syscall/mksyscall.pl: interpreter directive changed from "#!/usr/bin/env perl" to "/nix/store/dk5lnlbfz4vgyywifl4zaa988p9ixa2n-perl-5.34.1/bin/perl"
./src/syscall/mksysnum_netbsd.pl: interpreter directive changed from "#!/usr/bin/env perl" to "/nix/store/dk5lnlbfz4vgyywifl4zaa988p9ixa2n-perl-5.34.1/bin/perl"
./src/bootstrap.bash: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/iffl6dlplhv22i2xy7n1w51a5r631kmi-bash-5.1-p16/bin/bash"
./src/clean.bash: interpreter directive changed from "#!/usr/bin/env bash" to "/nix/store/iffl6dlplhv22i2xy7n1w51a5r631kmi-bash-5.1-p16/bin/bash"
./src/net/http/cgi/testdata/test.cgi: interpreter directive changed from "#!/usr/bin/perl" to "/nix/store/dk5lnlbfz4vgyywifl4zaa988p9ixa2n-perl-5.34.1/bin/perl"
./src/regexp/syntax/make_perl_groups.pl: interpreter directive changed from "#!/usr/bin/perl" to "/nix/store/dk5lnlbfz4vgyywifl4zaa988p9ixa2n-perl-5.34.1/bin/perl"
./lib/time/update.bash: interpreter directive changed from "#!/bin/bash" to "/nix/store/iffl6dlplhv22i2xy7n1w51a5r631kmi-bash-5.1-p16/bin/bash"
applying patch /nix/store/x48d0s4gns4jrck6qkwrpqn7nh9ygpx6-remove-tools-1.11.patch
patching file src/go/build/build.go
Hunk #1 succeeded at 1988 (offset 280 lines).
patching file src/runtime/extern.go
Hunk #1 succeeded at 250 with fuzz 2 (offset 21 lines).
applying patch /nix/store/1wxnz5npl05sahb5iv5q4azg2rk59kz5-ssl-cert-file-1.16.patch
patching file src/crypto/x509/root.go
Hunk #1 succeeded at 10 with fuzz 2 (offset 4 lines).
Hunk #2 succeeded at 28 (offset 4 lines).
applying patch /nix/store/hyif3rqf4kgbzbqkrv3bjspn0r8s5pl6-remove-test-pie-1.15.patch
patching file src/cmd/dist/test.go
Hunk #1 succeeded at 651 (offset 67 lines).
applying patch /nix/store/39bjxj20rygi2zlx6zscv38jxh2k7nkl-creds-test.patch
patching file src/syscall/creds_test.go
Hunk #1 succeeded at 78 (offset 2 lines).
applying patch /nix/store/bb1yc64b2c5pdnd2bgfzxccvzh0svz9f-go-1.9-skip-flaky-19608.patch
patching file src/net/http/serve_test.go
Hunk #1 succeeded at 5712 with fuzz 1 (offset 351 lines).
applying patch /nix/store/ddg24v72gz6vk7h9lvq1baz30bq1r777-skip-chown-tests-1.16.patch
patching file src/os/os_unix_test.go
Hunk #1 succeeded at 41 (offset 1 line).
Hunk #2 succeeded at 85 (offset 1 line).
Hunk #3 succeeded at 129 (offset 1 line).
applying patch /nix/store/4vqgx07v9xywvmgmdzk87561y734vm8n-skip-external-network-tests-1.16.patch
patching file src/internal/testenv/testenv.go
patching file src/net/dial_test.go
Hunk #1 succeeded at 970 (offset -20 lines).
applying patch /nix/store/hm2jzy93m5sl48alp2dhklz2b592dn9b-skip-nohup-tests.patch
patching file src/os/signal/signal_test.go
Hunk #1 succeeded at 292 (offset 75 lines).
Hunk #2 succeeded at 411 with fuzz 2 (offset 103 lines).
applying patch /nix/store/92qy0wn9289a1id4c5byc2mqc6v6xas2-skip-cgo-tests-1.15.patch
patching file src/cmd/dist/test.go
Hunk #1 succeeded at 1206 with fuzz 1 (offset 70 lines).
applying patch /nix/store/1cwv8hmzzn9s9b7rcvpymk6z6gm0c64p-go_no_vendor_checks-1.16.patch
patching file src/cmd/go/internal/modload/vendor.go
Hunk #1 succeeded at 138 (offset 5 lines).
applying patch /nix/store/amwcfcjv3bnmrfnfchhf8n60f93irkr9-skip-test-extra-files-on-386-1.14.patch
patching file src/os/exec/exec_test.go
Hunk #1 succeeded at 612 (offset -15 lines).
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
real-time non-blocking time  (microseconds, -R) unlimited
core file size              (blocks, -c) 0
data seg size               (kbytes, -d) unlimited
scheduling priority                 (-e) 0
file size                   (blocks, -f) unlimited
pending signals                     (-i) 63603
max locked memory           (kbytes, -l) 8192
max memory size             (kbytes, -m) unlimited
open files                          (-n) 4096
pipe size                (512 bytes, -p) 8
POSIX message queues         (bytes, -q) 819200
real-time priority                  (-r) 0
stack size                  (kbytes, -s) 8192
cpu time                   (seconds, -t) unlimited
max user processes                  (-u) 63603
virtual memory              (kbytes, -v) unlimited
file locks                          (-x) unlimited
@nix { "action": "setPhase", "phase": "buildPhase" }
building
no Makefile, doing nothing
Building Go cmd/dist using /nix/store/5s5bxly67d3axdssy4f9kfi0n1g2dwcd-go-bootstrap/share/go. (go1.16 linux/amd64)
Building Go toolchain1 using /nix/store/5s5bxly67d3axdssy4f9kfi0n1g2dwcd-go-bootstrap/share/go.
Building Go bootstrap cmd/go (go_bootstrap) using Go toolchain1.
Building Go toolchain2 using go_bootstrap and Go toolchain1.
Building Go toolchain3 using go_bootstrap and Go toolchain2.
Building packages and commands for linux/amd64.
---
Installed Go for linux/amd64 in /tmp/nix-build-go-1.17.10.drv-0/go
Installed commands in /tmp/nix-build-go-1.17.10.drv-0/go/bin

The binaries expect /tmp/nix-build-go-1.17.10.drv-0/go to be copied or moved to /nix/store/p3hzj4gp3pbd8cm1hpbk148s8sh21858-go-1.17.10/share/go
buildPhase completed in 4 minutes 13 seconds
@nix { "action": "setPhase", "phase": "checkPhase" }
running tests

##### Testing packages.
ok  	archive/tar	0.186s
ok  	archive/zip	2.855s
ok  	bufio	0.078s
ok  	bytes	7.060s
ok  	compress/bzip2	0.069s
ok  	compress/flate	4.393s
ok  	compress/gzip	0.672s
ok  	compress/lzw	0.136s
ok  	compress/zlib	1.230s
ok  	container/heap	0.172s
ok  	container/list	0.085s
ok  	container/ring	0.036s
ok  	context	0.026s
ok  	crypto	0.248s
ok  	crypto/aes	0.445s
ok  	crypto/cipher	0.014s
ok  	crypto/des	0.023s
ok  	crypto/dsa	0.214s
ok  	crypto/ecdsa	0.024s
ok  	crypto/ed25519	0.071s
ok  	crypto/ed25519/internal/edwards25519	9.267s
ok  	crypto/ed25519/internal/edwards25519/field	4.162s
ok  	crypto/elliptic	0.127s
ok  	crypto/elliptic/internal/fiat	0.018s
ok  	crypto/hmac	0.007s
ok  	crypto/internal/subtle	0.003s
ok  	crypto/md5	0.041s
ok  	crypto/rand	0.017s
ok  	crypto/rc4	0.056s
ok  	crypto/rsa	0.142s
ok  	crypto/sha1	0.046s
ok  	crypto/sha256	0.033s
ok  	crypto/sha512	0.026s
ok  	crypto/subtle	0.009s
ok  	crypto/tls	0.713s
ok  	crypto/x509	0.550s
ok  	database/sql	0.663s
ok  	database/sql/driver	0.060s
ok  	debug/dwarf	0.030s
ok  	debug/elf	0.055s
ok  	debug/gosym	0.048s
ok  	debug/macho	0.018s
ok  	debug/pe	0.011s
ok  	debug/plan9obj	0.010s
ok  	embed/internal/embedtest	0.003s
ok  	encoding/ascii85	0.021s
ok  	encoding/asn1	0.011s
ok  	encoding/base32	0.019s
ok  	encoding/base64	0.030s
ok  	encoding/binary	0.061s
ok  	encoding/csv	0.035s
ok  	encoding/gob	0.058s
ok  	encoding/hex	0.024s
ok  	encoding/json	2.647s
ok  	encoding/pem	19.298s
ok  	encoding/xml	0.127s
ok  	errors	0.050s
ok  	expvar	0.014s
ok  	flag	0.482s
ok  	fmt	0.125s
ok  	go/ast	0.089s
ok  	go/build	5.201s
ok  	go/build/constraint	0.098s
ok  	go/constant	0.005s
ok  	go/doc	0.448s
ok  	go/format	0.086s
ok  	go/importer	1.145s
ok  	go/internal/gccgoimporter	0.032s
ok  	go/internal/gcimporter	8.045s
ok  	go/internal/srcimporter	32.711s
ok  	go/parser	0.062s
ok  	go/printer	0.872s
ok  	go/scanner	0.067s
ok  	go/token	0.028s
ok  	go/types	57.478s
ok  	hash	0.016s
ok  	hash/adler32	0.010s
ok  	hash/crc32	0.107s
ok  	hash/crc64	0.029s
ok  	hash/fnv	0.086s
ok  	hash/maphash	0.671s
ok  	html	0.452s
ok  	html/template	0.139s
ok  	image	0.121s
ok  	image/color	0.042s
ok  	image/draw	0.079s
ok  	image/gif	0.427s
ok  	image/jpeg	0.261s
ok  	image/png	0.050s
ok  	index/suffixarray	0.188s
ok  	internal/abi	0.242s
ok  	internal/cpu	0.064s
ok  	internal/execabs	0.008s
ok  	internal/fmtsort	0.241s
ok  	internal/itoa	0.029s
ok  	internal/poll	0.124s
ok  	internal/profile	0.023s
ok  	internal/reflectlite	0.366s
ok  	internal/singleflight	0.041s
ok  	internal/trace	0.095s
ok  	internal/unsafeheader	0.013s
ok  	internal/xcoff	0.658s
ok  	io	0.176s
ok  	io/fs	0.238s
ok  	io/ioutil	0.395s
ok  	log	0.141s
ok  	log/syslog	4.076s
ok  	math	0.011s
ok  	math/big	1.395s
ok  	math/bits	0.010s
ok  	math/cmplx	0.019s
ok  	math/rand	0.560s
ok  	mime	0.037s
ok  	mime/multipart	0.251s
ok  	mime/quotedprintable	0.028s
ok  	net	12.266s
ok  	net/http	19.740s
ok  	net/http/cgi	0.730s
ok  	net/http/cookiejar	0.007s
ok  	net/http/fcgi	0.114s
ok  	net/http/httptest	0.059s
ok  	net/http/httptrace	0.080s
ok  	net/http/httputil	1.255s
ok  	net/http/internal	0.053s
ok  	net/http/internal/ascii	0.049s
ok  	net/http/pprof	5.673s
ok  	net/internal/socktest	0.023s
ok  	net/mail	0.011s
ok  	net/rpc	0.408s
ok  	net/rpc/jsonrpc	0.202s
ok  	net/smtp	0.161s
ok  	net/textproto	0.133s
ok  	net/url	0.212s
ok  	os	5.248s
ok  	os/exec	6.032s
ok  	os/signal	6.418s
ok  	os/user	0.197s
ok  	path	0.003s
ok  	path/filepath	0.674s
ok  	plugin	0.047s
ok  	reflect	0.503s
ok  	regexp	0.537s
ok  	regexp/syntax	0.543s
ok  	runtime	388.549s
ok  	runtime/cgo	0.007s
ok  	runtime/debug	0.069s
ok  	runtime/internal/atomic	0.041s
ok  	runtime/internal/math	0.026s
ok  	runtime/internal/sys	0.030s
ok  	runtime/metrics	0.057s
ok  	runtime/pprof	97.699s
ok  	runtime/trace	2.300s
ok  	sort	0.414s
ok  	strconv	3.651s
ok  	strings	4.346s
ok  	sync	0.533s
ok  	sync/atomic	1.926s
ok  	syscall	2.816s
ok  	testing	1.469s
ok  	testing/fstest	0.005s
ok  	testing/iotest	0.192s
ok  	testing/quick	0.190s
ok  	text/scanner	0.008s
ok  	text/tabwriter	0.048s
ok  	text/template	0.423s
ok  	text/template/parse	0.108s
ok  	time	13.740s
ok  	unicode	0.013s
ok  	unicode/utf16	0.070s
ok  	unicode/utf8	0.178s
ok  	cmd/addr2line	46.725s
ok  	cmd/api	73.246s
ok  	cmd/asm/internal/asm	11.391s
ok  	cmd/asm/internal/lex	0.350s
ok  	cmd/compile/internal/dwarfgen	1.932s
ok  	cmd/compile/internal/importer	24.176s
ok  	cmd/compile/internal/ir	0.039s
ok  	cmd/compile/internal/logopt	1.935s
ok  	cmd/compile/internal/noder	0.255s
ok  	cmd/compile/internal/ssa	1.892s
ok  	cmd/compile/internal/ssagen	0.040s
ok  	cmd/compile/internal/syntax	0.106s
ok  	cmd/compile/internal/test	140.827s
ok  	cmd/compile/internal/typecheck	4.193s
ok  	cmd/compile/internal/types	0.959s
ok  	cmd/compile/internal/types2	8.576s
ok  	cmd/cover	11.676s
ok  	cmd/doc	1.959s
ok  	cmd/fix	9.087s
go test proxy running at GOPROXY=http://127.0.0.1:45511/mod
--- FAIL: TestScript (0.01s)
    --- FAIL: TestScript/mod_find (1.97s)
        script_test.go:253: 
            # Derive module path from import comment. (0.792s)
            # Import comment works even with CRLF line endings. (0.106s)
            # Derive module path from location inside GOPATH.
            # 'go mod init' should succeed if modules are not explicitly disabled. (0.225s)
            # go mod init rejects a zero-length go.mod file (0.082s)
            # Module path from Godeps/Godeps.json overrides GOPATH. (0.077s)
            # Empty directory outside GOPATH fails. (0.144s)
            # Empty directory inside GOPATH/src uses location inside GOPATH. (0.353s)
            # In Plan 9, directories are automatically created in /n.
            # For example, /n/go.mod always exist, but it's a directory.
            # Test that we ignore directories when trying to find go.mod. (0.127s)
            > cd $WORK/gomoddir
            $WORK/gomoddir
            > ! go list .
            [stderr]
            go: cannot find main module, but found .git/config in /
            	to create a module there, run:
            	cd ../../../../../.. && go mod init
            [exit status 1]
            > stderr '^go: go.mod file not found in current directory or any parent directory; see ''go help modules''$'
            FAIL: testdata/script/mod_find.txt:52: no match for `(?m)^go: go.mod file not found in current directory or any parent directory; see 'go help modules'$` found in stderr
            
    --- FAIL: TestScript/mod_convert_dep (0.75s)
        script_test.go:253: 
            # We should not create a go.mod file unless the user ran 'go mod init' explicitly.
            # However, we should suggest 'go mod init' if we can find an alternate config file. (0.101s)
            # The command we suggested should succeed. (0.221s)
            # In Plan 9, directories are automatically created in /n.
            # For example, /n/Gopkg.lock always exists, but it's a directory.
            # Test that we ignore directories when trying to find alternate config files. (0.272s)
            > cd $WORK/gopkgdir/x
            $WORK/gopkgdir/x
            > ! go list .
            [stderr]
            go: cannot find main module, but found .git/config in /
            	to create a module there, run:
            	cd ../../../../../../.. && go mod init
            [exit status 1]
            > stderr '^go: go.mod file not found in current directory or any parent directory; see ''go help modules''$'
            FAIL: testdata/script/mod_convert_dep.txt:21: no match for `(?m)^go: go.mod file not found in current directory or any parent directory; see 'go help modules'$` found in stderr
            
FAIL
FAIL	cmd/go	239.159s
ok  	cmd/go/internal/auth	0.023s
ok  	cmd/go/internal/cache	0.215s
ok  	cmd/go/internal/fsys	0.169s
ok  	cmd/go/internal/generate	0.020s
ok  	cmd/go/internal/get	0.115s
ok  	cmd/go/internal/imports	0.042s
ok  	cmd/go/internal/load	0.088s
ok  	cmd/go/internal/lockedfile	0.186s
ok  	cmd/go/internal/lockedfile/internal/filelock	0.082s
ok  	cmd/go/internal/modconv	0.079s
ok  	cmd/go/internal/modfetch	0.014s
ok  	cmd/go/internal/modfetch/codehost	0.009s
ok  	cmd/go/internal/modfetch/zip_sum_test	0.013s
ok  	cmd/go/internal/modload	0.038s
ok  	cmd/go/internal/mvs	0.008s
ok  	cmd/go/internal/par	0.026s
ok  	cmd/go/internal/search	0.023s
ok  	cmd/go/internal/str	0.018s
ok  	cmd/go/internal/test	0.023s
ok  	cmd/go/internal/txtar	0.020s
ok  	cmd/go/internal/vcs	0.014s
ok  	cmd/go/internal/web	0.020s
ok  	cmd/go/internal/work	0.239s
ok  	cmd/gofmt	0.163s
ok  	cmd/internal/archive	73.779s
ok  	cmd/internal/buildid	0.426s
ok  	cmd/internal/dwarf	0.251s
ok  	cmd/internal/edit	0.115s
ok  	cmd/internal/goobj	0.150s
ok  	cmd/internal/moddeps	71.948s
ok  	cmd/internal/obj	0.154s
ok  	cmd/internal/obj/ppc64	1.872s
ok  	cmd/internal/obj/riscv	1.416s
ok  	cmd/internal/obj/s390x	0.091s
ok  	cmd/internal/obj/x86	1.849s
ok  	cmd/internal/objabi	0.205s
ok  	cmd/internal/pkgpath	0.013s
ok  	cmd/internal/src	0.108s
ok  	cmd/internal/sys	0.057s
ok  	cmd/internal/test2json	0.322s
ok  	cmd/link	59.598s
ok  	cmd/link/internal/benchmark	0.109s
ok  	cmd/link/internal/ld	38.804s
ok  	cmd/link/internal/loader	0.181s
ok  	cmd/nm	39.362s
ok  	cmd/objdump	45.939s
ok  	cmd/pack	29.757s
ok  	cmd/pprof	13.128s
ok  	cmd/trace	0.684s
ok  	cmd/vet	75.042s
FAIL
go tool dist: Failed: exit status 1
```

**Comment by winterqt** [comment]
@shadowrylander What Nixpkgs commit are you on? Looks like an old one from the fact that it's trying to build Go 1.17.10.

**Comment by shadowrylander** [comment]
My default is set to `nixos-22.05`; which one should I be using? I copied over the derivation and used it in a `callPackage`, by the way.

**Comment by winterqt** [comment]
Oh, that makes sense.

Go 1.17.10 is [green on Hydra](https://hydra.nixos.org/build/184715337) for 22.05 -- can you run `[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here` and paste the output here?

**Comment by shadowrylander** [comment]
I'm running this from a flake; does that make a difference?
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.16.11, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.10.0pre20220706_f6316b4`
 - channels(shadowrylander): `"j"`
 - channels(root): `"j, nixos-21.11.336282.ed02c2ba038"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by winterqt** [comment]
Actually, yes, it doesn't give me the required info. 😅

Assuming [this](https://github.com/shadowrylander/shadowrylander/blob/62e2c045cd70df692a8dcc1bc280c5b4ca4cd029/etc/nixos/flake.nix) is the flake in question, can you please provide a diff or similar so I can see how you're trying to use this package (ideally so I can reproduce your results)?

**Comment by shadowrylander** [comment]
My overlays and callPackages not specific to my nixos config are here: https://github.com/sylvorg/settings/blob/main/flake.nix#L727

I just used `final.callPackage` on it; it was pretty much an exact copy of the current derivation here!

**Comment by winterqt** [comment]
Uhm, that's a lot of code. Can you please provide me the following info, which I wasn't able to grasp by grepping around: which Nixpkgs input is used to `callPackage` this?

**Comment by shadowrylander** [comment]
Sorry; I guess more modularity means more confusion! 😅 The `inputs.nixpkgs` [here](https://github.com/sylvorg/settings/blob/main/flake.nix#L1917) is declared [here](https://github.com/sylvorg/settings/blob/main/flake.nix#L44)!

**Comment by winterqt** [comment]
Go 1.17.10 for `x86_64-linux` on d9536b9b1e33377fb520e115bc8588d093ed97be (your `nixpkgs` input) corresponds to `/nix/store/cy0yxszv2ny0dsybz6j786x9rmajai9v-go-1.17.10`, but it looks like you're building `/nix/store/p3hzj4gp3pbd8cm1hpbk148s8sh21858-go-1.17.10`. Clearly there's *something* weird happening here, but I'm really not sure what.

If you can create a minimal example that I can use to reproduce the issue, I'd be happy to take a look. But right now I'm not sure at all what could be going on here.

**Comment by shadowrylander** [comment]
Oh... Could it be because I'm patching nixpkgs? There's a specific nixos module from the official repos that I need to modify.

**Comment by winterqt** [comment]
Running `nix build .#gum` on `x86_64-linux` is causing a lot of things to be built from source that I can't reproduce on d9536b9b1e33377fb520e115bc8588d093ed97be, or when building from your flake for `aarch64-darwin`. Maybe that info will help you narrow things down a bit?

**Comment by shadowrylander** [comment]
Hmm... Honestly, the only other Go related thing I have in the flake is a custom `caddy` build, which shouldn't be affecting it... Either way, I've switched to using the following overlay:
```nix
final: prev: { gum = prev.gum or inputs.nixos-master.legacyPackages.${final.stdenv.targetPlatform.system}.gum; }
```
Thanks for trying, though!

**Comment by winterqt** [comment]
The thing is Caddy can't build either, if your Go build is failing. So this is just... very, *very* weird.

**Comment by shadowrylander** [comment]
Ah, right... I'd forgetten about that... Do you think I should select a particular version to build, such as specifically `buildGo117Module` or similar?

**Comment by winterqt** [comment]
Well Go 1.17's build is failing, so...

I suggest you narrow down your overlay usage to see what could possibly be going wrong here, since there's definitely something that isn't right as evidenced by the Go 1.17 build being different than it should be on the Nixpkgs revision you use.

Also, we should probably stop discussing this here to avoid emailing so many people. Feel free to pop by the Matrix channel and find me, we can talk more there if you'd like.

**Comment by shadowrylander** [comment]
Fair enough! I'll see what I can do.

Also, matrix channels were always a little confusing to me; I'll see if I can find you sometime! 😅

**Pull request merged**